server/models/Request.js

```diff
@@ -0,0 +1,28 @@
+const mongoose = require("mongoose");
+
+const requestSchema = new mongoose.Schema({
+  user_id: { type: String, max: 50, required: true },
+  sitter_id: { type: String, max: 50, required: true },
+  pet_ids: { type: [String], max: 50, required: true },
+  start_date: {
+    type: Date,
+    validate: {
+		validator: function(startDate) { return startDate < this.end_date },
+		message: 'Start date must begin before the end date.'
+    },
+	required: [true, 'Must have a start date.']
+  },
+  end_date: {
+    type: Date,
+    validate: {
+		validator: function(endDate) { return this.start_date < endDate },
+		message: 'Start date must begin before the end date.'
+    },
+	required: [true, 'Must have an end date.']
+  },
+  accepted: { type: Boolean, default: False },
+  declined: { type: Boolean, default: False },
```

Comment:
Having multiple fields that needs to be synchronized with each other usually leads to unnecessary complexity - can we simplify these into a single field (e.g. `status` with some [string `SchemaTypes`](https://mongoosejs.com/docs/schematypes.html#string-validators)) ?

Reply:
So, would a status field with a String enum with 'accepted' and 'declined' be appropriate?

Reply:
Yes, that would be a good idea. We can have `['pending', 'accepted', 'declined', 'completed']` as the options for the status of a request. As for the `paid` field we can have it seperate from the status as we only want it to be accessed through the payment routes 👍🏽